src/Microsoft.SqlTools.ServiceLayer/Admin/Contracts/DatabaseInfoWrapper.cs

```diff
@@ -0,0 +1,67 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SqlTools.ServiceLayer.Admin.Contracts 
+{
+    public class DatabaseInfoWrapper
+    {
+        public string name;
+        public string owner;
+        public string collation;
+        public string recoveryModel;
+        public string databaseState;
+        
+        public DatabaseInfoWrapper(DatabaseInfo info)
+        {
+            object placeholder = null;
+            
+            info.Options.TryGetValue(AdminServicesProviderOptionsHelper.Name, out placeholder);
+            
+            if (placeholder != null)
+            {
+                name = (string) placeholder;
```

Comment:
This can cause ClassCastExceptions. Instead please use `placeholder as string`. If it's incompatible it'll be null.